materialize-sql-json/driver.go

```diff
@@ -22,91 +27,95 @@ type Driver struct {
 	ResourceSpecType Resource
 	// NewEndpoint returns an *Endpoint which will be used to handle interactions with the database.
 	NewEndpoint func(_ context.Context, endpointConfig json.RawMessage) (*Endpoint, error)
+	// Transactor to delegate Load and Store to.
+	transactor Transactor
 }
 
-var _ pm.DriverServer = &Driver{}
+var _ boilerplate.MaterializeDriver = &Driver{}
 
-// Spec implements the DriverServer interface.
-func (d *Driver) Spec(ctx context.Context, req *pm.SpecRequest) (*pm.SpecResponse, error) {
+func (d *Driver) Spec(ctx context.Context, req protocol.SpecRequest) (*protocol.SpecResponse, error) {
 	var endpoint, resource []byte
+	var err error
 
-	if err := req.Validate(); err != nil {
```

Comment:
Why are we removing these `Validate()` calls?

Reply:
The protobuf types provided comprehensive high-level validation methods that were called here and elsewhere. The paradigm going forward is to trust that data we get from the runtime is correct, since it is responsible for ensuring protocol correctness. The runtime can do these extensive validations in a higher leverage way than every connector needing to implement the validations for its JSON protocol types.

Reply:
This is in a bit of flux right now, FYI. As of yesterday, there's a pathway where the protobuf protocol and the JSON protocol are one-and-the-same: the JSON connector protocol is quite literally the canonical JSON encoding of the protobuf types.

That would make it practical to have Go connectors use the generated Go package for protobuf types and decode them as JSON, which would also allow the Validate methods to come along.

Apologies for the back and forth, I've been actively working on figuring this all out.

Reply:
None of this is a retraction of the position that connectors can essentially trust what they receive from the runtime, by the way, but it would allow connectors to continue to call Validate as an extra sanity check. I'm not taking a position on whether they _should_.

Reply:
That does seem like some interesting flux, so I'll certainly wait and see how that turns out before I merge this, and probably make some changes here it sounds like. If I'm understanding what you are saying correctly this would obviate the need for a bespoke set of Go JSON structs as I have included here, which would certainly be nice.

I still prefer the idea of the trusting the input from the runtime and not validating it as a rule in connectors. Otherwise we're going to be in an ambiguous place where we are always making sure we are doing something that is largely redundant.

There may be selective cases where a validate method would be handy though. One example is pulling the persisted spec from the SQL endpoint. It could be argued that validating that is not totally redundant since it could have been changed in the table somehow.